shell/plugins/dashboard-store/mutations.js

```diff
@@ -98,37 +98,52 @@ export function load(state, {
 
   cache.generation++;
 
-  let entry;
+  let entry = cache.map.get(id);
+  const inMap = !!entry;
 
+  //
```

Comment:
I've simplified this whole section
Step 1 - Determine the `entry` that should be in the local map and list cache
- Ensure that this entry is the latest and greatest
- Ensure we merge the latest and greatest into the reference in the local map/list
Step 2 - Ensure the `entry` is in both both map and list
- Goes on the assumption parity between map and list

Reply:
It really seems like we should have a custom class (combine the list and map) that handles these patterns for us. 

Probably a better fit for a future change.

Reply:
Strongly agree on both parts! Something that's performant and hides the complexity would be grand. Definitely one for the future